test/robot/web/www/models/robotModel.js

```diff
@@ -0,0 +1,459 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
Maybe not one for now, but one of the nice things about implementing this in javascript is that we can sensibly use [protobufs now](https://developers.google.com/protocol-buffers/docs/reference/javascript-generated), and not have to use strings for field names.